docs/en/table_design/indexes/Ngram_Bloom_Filter_Index.md

```diff
@@ -0,0 +1,128 @@
+- N-gram bloom filter indexes can only be created for columns of the string types (CHAR, STRING, and VARCHAR).
+- To determine whether a query hits an N-gram bloom filter index, you can check the `BloomFilterFilterRows` field in the query's profile.
+- Only one type of index (Bloom Filter or N-gram Bloom Filter) can be created for a single column.
+
+## Basic operation 
+
+### Create N-gram bloom filter index
+
+```SQL
+CREATE TABLE test.table1
+(
+    k1 CHAR(10),
+    k2 CHAR(10),
+    v1 INT SUM,
+    INDEX index_name (k2) USING NGRAMBF ("gram_num" = "4",
+                                         "bloom_filter_fpp" = "0.05") COMMENT ''
+)
+ENGINE = olap
+AGGREGATE KEY(k1, k2)
+DISTRIBUTED BY HASH(k1);
+
+```
+
+N-gram bloom filter index related parameters:
+
+| **Parameter**    | **Required** | **Description**                                              |
+| ---------------- | ------------ | ------------------------------------------------------------ |
+| `index_name`       | Yes          | The name of the index. Index names must be unique within a table. |
+| `column_name`      | Yes          | The name of the column for which the index is created. Only a single column name can be specified. In the example above, it is `k2`. |
+| `gram_num`         | Yes          | The length of a substring after the string in the indexed column is tokenized. In the example above, `gram_num` is `4`. |
+| `bloom_filter_fpp` | No           | The false positive possibility of the Bloom filter, ranging from 0.0001 to 0.05. The default value is 0.05. A smaller value provides better filtering but incurs greater storage overhead. |
+| `case_sensitive`   |  No          | Whether this index is case-sensitive or not. Default value is `case_sensitive`. |
+| `COMMENT`          | No           | Index comment. |
+
+For explanations of other parameters related to table creation, see [CREATE TABLE](../../sql-reference/sql-statements/data-definition/CREATE_TABLE.md).
+
+### View N-gram bloom filter index 
+
+You can view all indexes of a table using `SHOW CREATE TABLE` or `SHOW INDEX FROM` table. Since index creation is asynchronous, you can only see the corresponding index after the index is successfully created.
+
+```SQL
+SHOW CREATE TABLE table1;
+SHOW INDEX FROM table1;
+```
+
+### Modify N-gram bloom filter index
+
+You can add and delete the N-gram bloom filter index by using the [ALTER TABLE](../../sql-reference/sql-statements/data-definition/ALTER_TABLE.md) statement.
+
+- Execute the following statement to add a new N-gram bloom filter index `new_index_name` for the column `k1` to the table `table1`.
+
+  ```SQL
+  ALTER TABLE table1 
+  ADD INDEX new_index_name(k1) USING NGRAMBF ("gram_num" = "4", 
+                                              "bloom_filter_fpp" = "0.05") COMMENT '';
+  ```
+
+- The following statement to remove the N-gram bloom filter index `new_index_name` from the table `table1`.
```

Comment:
Execute the following statement to remove 

or

The following statement is to remove